scripts/generate-conda-envs.py

```diff
@@ -0,0 +1,241 @@
+# Copyright (c) 2020-2022, NVIDIA CORPORATION. All rights reserved.
+#
+# NVIDIA CORPORATION and its licensors retain all intellectual property
+# and proprietary rights in and to this software, related documentation
+# and any modifications thereto.  Any use, reproduction, disclosure or
+# distribution of this software and related documentation without an express
+# license agreement from NVIDIA CORPORATION is strictly prohibited.
+#
+# See the LICENSE file for details.
+#
+from __future__ import annotations
+
+from dataclasses import dataclass
+from typing import Literal, Protocol
+
+from jinja2 import Template
+from typing_extensions import TypeAlias
+
+# --- Types -------------------------------------------------------------------
+
+Req: TypeAlias = str
+Reqs: TypeAlias = tuple[Req, ...]
+OSType: TypeAlias = Literal["linux", "darwin"]
+
+
+class SectionConfig(Protocol):
+    heading: str
+
+    @property
+    def conda(self) -> Reqs:
+        return ()
+
+    @property
+    def pip(self) -> Reqs:
+        return ()
+
+
+@dataclass(frozen=True)
+class CUDAConfig(SectionConfig):
+    ctk_version: str | None
+
+    header = "cuda"
+
+    @property
+    def conda(self) -> Reqs:
+        if self.ctk_version is None:
+            return ()
+
+        return (
+            f"cudatoolkit=={self.ctk_version}",  # runtime
+            "cutensor",  # runtime
+            "nccl",  # runtime
+            "pynvml",  # tests
+        )
+
+    @property
+    def filename_component(self) -> str:
+        if self.ctk_version is None:
+            return ""
+
+        return f"-cuda-{self.ctk_version}"
+
+
+class BuildConfig(SectionConfig):
+    header = "build"
+
+    @property
+    def conda(self) -> Reqs:
+        return (
+            "c-compiler",
+            "cmake>=3.24",
+            "cxx-compiler",
+            "gcc_linux-64 # [linux64]",
```

Comment:
Let's remove this requirement for `gcc` specifically. On linux `c-compiler` will automatically pull gcc anyway, and on osx this package doesn't exist (`c-compiler` will pull clang I believe). CC @m3vaz